Importing module 'gym_37' (/home/mmpug/ws/IsaacGym_Preview_4_Package/isaacgym/python/isaacgym/_bindings/linux-x86_64/gym_37.so)
Setting GYM_USD_PLUG_INFO_PATH to /home/mmpug/ws/IsaacGym_Preview_4_Package/isaacgym/python/isaacgym/_bindings/linux-x86_64/usd/plugInfo.json
PyTorch version 1.8.1
Device count 1
/home/mmpug/ws/IsaacGym_Preview_4_Package/isaacgym/python/isaacgym/_bindings/src/gymtorch
Using /home/mmpug/.cache/torch_extensions as PyTorch extensions root...
Emitting ninja build file /home/mmpug/.cache/torch_extensions/gymtorch/build.ninja...
Building extension module gymtorch...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
ninja: no work to do.
Loading extension module gymtorch...
2023-10-22 20:16:54,471 - INFO - logger - logger initialized
<unknown>:6: DeprecationWarning: invalid escape sequence \*
Error: FBX library failed to load - importing FBX data will not succeed. Message: No module named 'fbx'
FBX tools must be installed from https://help.autodesk.com/view/FBX/2020/ENU/?guid=FBX_Developer_Help_scripting_with_python_fbx_installing_python_fbx_html
/home/mmpug/Eureka/eureka/../isaacgymenvs/isaacgymenvs/train.py:75: UserWarning: 
The version_base parameter is not specified.
Please specify a compatability version level, or None.
Will assume defaults for version 1.1
  @hydra.main(config_name="config", config_path="./cfg")
/home/mmpug/anaconda3/envs/rlgpu/lib/python3.7/site-packages/hydra/_internal/defaults_list.py:415: UserWarning: In config: Invalid overriding of hydra/job_logging:
Default list overrides requires 'override' keyword.
See https://hydra.cc/docs/1.2/upgrades/1.0_to_1.1/defaults_list_override for more information.

  deprecation_warning(msg)
/home/mmpug/anaconda3/envs/rlgpu/lib/python3.7/site-packages/hydra/_internal/hydra.py:127: UserWarning: Future Hydra versions will no longer change working directory at job runtime by default.
See https://hydra.cc/docs/1.2/upgrades/1.1_to_1.2/changes_to_job_working_dir/ for more information.
  configure_logging=with_log_configuration,
/home/mmpug/anaconda3/envs/rlgpu/lib/python3.7/site-packages/torch/utils/tensorboard/__init__.py:3: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  if not hasattr(tensorboard, '__version__') or LooseVersion(tensorboard.__version__) < LooseVersion('1.15'):
Setting seed: 42
Network Directory: /home/mmpug/Eureka/eureka/outputs/eureka/2023-10-22_20-10-01/policy-2023-10-22_20-16-55/runs/ShadowHandGPT-2023-10-22_20-16-55/nn
Tensorboard Directory: /home/mmpug/Eureka/eureka/outputs/eureka/2023-10-22_20-10-01/policy-2023-10-22_20-16-55/runs/ShadowHandGPT-2023-10-22_20-16-55/summaries
self.seed = 42
Started to train
Error executing job with overrides: ['task=ShadowHandGPT', 'wandb_activate=False', 'wandb_entity=', 'wandb_project=', 'headless=True', 'capture_video=False', 'force_render=False', 'max_iterations=3000']
Traceback (most recent call last):
  File "/home/mmpug/Eureka/isaacgymenvs/isaacgymenvs/utils/rlgames_utils.py", line 116, in create_rlgpu_env
    module = importlib.import_module(module_name)
  File "/home/mmpug/anaconda3/envs/rlgpu/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/mmpug/Eureka/isaacgymenvs/isaacgymenvs/tasks/shadow_handgpt.py", line 767, in <module>
    object_angvel: torch.Tensor) -> Tuple[torch.Tensor, Dict[str, torch.Tensor]]:
  File "/home/mmpug/anaconda3/envs/rlgpu/lib/python3.7/site-packages/torch/jit/_script.py", line 990, in script
    qualified_name, ast, _rcb, get_default_args(obj)
RuntimeError: 
undefined value ang_vel_penalty:
  File "/home/mmpug/Eureka/isaacgymenvs/isaacgymenvs/tasks/shadow_handgpt.py", line 784
    angvel_norm = torch.norm(object_angvel, dim=1)
    ang_vel_reward = torch.exp(-angvel_norm / ang_vel_temperature)
    ang_vel_penalty = torch.where(angvel_norm > ang_vel_threshold, torch.tensor(1.0).to(ang_vel_penalty.device), torch.tensor(0.0).to(ang_vel_penalty.device))
                                                                                        ~~~~~~~~~~~~~~~ <--- HERE

    # Combine rewards and penalties


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mmpug/Eureka/eureka/../isaacgymenvs/isaacgymenvs/train.py", line 207, in launch_rlg_hydra
    'sigma': cfg.sigma if cfg.sigma != '' else None
  File "/home/mmpug/Eureka/rl_games/rl_games/torch_runner.py", line 124, in run
    self.run_train(args)
  File "/home/mmpug/Eureka/rl_games/rl_games/torch_runner.py", line 98, in run_train
    self.agent = self.algo_factory.create(self.algo_name, base_name='run', params=self.params)
  File "/home/mmpug/Eureka/rl_games/rl_games/common/object_factory.py", line 15, in create
    return builder(**kwargs)
  File "/home/mmpug/Eureka/rl_games/rl_games/torch_runner.py", line 37, in <lambda>
    self.algo_factory.register_builder('a2c_continuous', lambda **kwargs : a2c_continuous.A2CAgent(**kwargs))
  File "/home/mmpug/Eureka/rl_games/rl_games/algos_torch/a2c_continuous.py", line 16, in __init__
    a2c_common.ContinuousA2CBase.__init__(self, base_name, params)
  File "/home/mmpug/Eureka/rl_games/rl_games/common/a2c_common.py", line 1076, in __init__
    A2CBase.__init__(self, base_name, params)
  File "/home/mmpug/Eureka/rl_games/rl_games/common/a2c_common.py", line 121, in __init__
    self.vec_env = vecenv.create_vec_env(self.env_name, self.num_actors, **self.env_config)
  File "/home/mmpug/Eureka/rl_games/rl_games/common/vecenv.py", line 222, in create_vec_env
    return vecenv_config[vec_env_name](config_name, num_actors, **kwargs)
  File "/home/mmpug/Eureka/eureka/../isaacgymenvs/isaacgymenvs/train.py", line 161, in <lambda>
    vecenv.register('RLGPU', lambda config_name, num_actors, **kwargs: RLGPUEnv(config_name, num_actors, **kwargs))
  File "/home/mmpug/Eureka/isaacgymenvs/isaacgymenvs/utils/rlgames_utils.py", line 253, in __init__
    self.env = env_configurations.configurations[config_name]['env_creator'](**kwargs)
  File "/home/mmpug/Eureka/eureka/../isaacgymenvs/isaacgymenvs/train.py", line 145, in <lambda>
    'env_creator': lambda **kwargs: create_isaacgym_env(**kwargs),
  File "/home/mmpug/Eureka/eureka/../isaacgymenvs/isaacgymenvs/train.py", line 123, in create_isaacgym_env
    **kwargs,
  File "/home/mmpug/Eureka/isaacgymenvs/isaacgymenvs/__init__.py", line 59, in make
    return create_rlgpu_env()
  File "/home/mmpug/Eureka/isaacgymenvs/isaacgymenvs/utils/rlgames_utils.py", line 119, in create_rlgpu_env
    task_caller = isaacgym_task_map[task_name]
KeyError: 'ShadowHandGPT'

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
